### ISSUE TITLE
Schain config can't be generated 

### ISSUE THREAD

**Issue by yatsunastya** [skalenetwork/skale-admin#973]
**Admin:** 2.5.1-develop.4

**Precondition**
https://github.com/skalenetwork/skale-admin/issues/971
https://github.com/skalenetwork/skale-admin/issues/972
There is schain `submissive-well-off-rotanev` with 5 rounds of failed DKG (the last one is also failed so there is 0 node in the group).

**To Reproduce**
1. Send `restartSchainCreation` transaction on `submissive-well-off-rotanev`
2. DKG starts (first round is failed because of `complaint bad data` [transaction](https://goerli.etherscan.io/tx/0x1d33d2c9a3d7902d9c25b7012089475582643768b0a76989fbd4aec9581da42b)
3. Second round passed. DKG is successful 
4. Go to admin logs on any node 
5. Pay attention to schain config generation

**Expected behavior:**
Schain config is successfully generated with correct rotation history

**Actual result:**
Admin collects all rotation history but fails at the moment of generation
```
[2023-09-05 11:57:07,114 INFO][24510][submissive-well-off-rotanev][MainThread] - skale.contracts.manager.node_rotation:115 - current_ts: 1693915020, schain_finish_ts: 1693742124
[2023-09-05 11:57:08,250 INFO][24510][submissive-well-off-rotanev][MainThread] - core.schains.monitor.main:183 - sync_config_run False, config_version 0.0.0, stream_version regression-2.2
[2023-09-05 11:57:08,250 INFO][24510][submissive-well-off-rotanev][MainThread] - core.schains.monitor.main:189 - Adding config task to the pool
[2023-09-05 11:57:08,251 INFO][24510][submissive-well-off-rotanev][MainThread] - core.schains.task:40 - Task submissive-well-off-rotanev-config finished with None
[2023-09-05 11:57:08,251 INFO][24510][submissive-well-off-rotanev][MainThread] - core.schains.task:42 - Running task submissive-well-off-rotanev-config
[2023-09-05 11:57:08,251 INFO][24510][submissive-well-off-rotanev][MainThread] - core.schains.monitor.main:157 - Monitor iteration completed, sleeping for 23
[2023-09-05 11:57:10,223 INFO][24510][submissive-well-off-rotanev][T_0] - core.schains.monitor.main:103 - Config checks: {'config_dir': True, 'dkg': True, 'external_state': True, 'upstream_config': False}
[2023-09-05 11:57:10,223 INFO][24510][submissive-well-off-rotanev][T_0] - core.schains.monitor.config_monitor:46 - Config monitor type RegularConfigMonitor starting
[2023-09-05 11:57:10,226 INFO][24510][submissive-well-off-rotanev][T_0] - web.models.schain:152 - Changing monitor_last_seen for submissive-well-off-rotanev to 2023-09-05 11:57:10.226933
[2023-09-05 11:57:10,233 INFO][24510][submissive-well-off-rotanev][T_0] - core.schains.monitor.action:190 - Creating new upstream_config rotation_id: 8, stream: regression-2.2
[2023-09-05 11:57:10,236 INFO][24510][submissive-well-off-rotanev][T_0] - core.schains.config.main:49 - Generating sChain config for submissive-well-off-rotanev
[2023-09-05 11:57:32,410 INFO][24510][submissive-well-off-rotanev][MainThread] - skale.contracts.manager.node_rotation:115 - current_ts: 1693915044, schain_finish_ts: 1693742124
[2023-09-05 11:57:33,595 INFO][24510][submissive-well-off-rotanev][MainThread] - core.schains.monitor.main:183 - sync_config_run False, config_version 0.0.0, stream_version regression-2.2
[2023-09-05 11:57:33,595 INFO][24510][submissive-well-off-rotanev][MainThread] - core.schains.monitor.main:189 - Adding config task to the pool
[2023-09-05 11:57:33,595 INFO][24510][submissive-well-off-rotanev][MainThread] - core.schains.monitor.main:157 - Monitor iteration completed, sleeping for 39
[2023-09-05 11:58:13,556 INFO][24510][submissive-well-off-rotanev][MainThread] - skale.contracts.manager.node_rotation:115 - current_ts: 1693915092, schain_finish_ts: 1693742124
[2023-09-05 11:58:14,721 INFO][24510][submissive-well-off-rotanev][MainThread] - core.schains.monitor.main:183 - sync_config_run False, config_version 0.0.0, stream_version regression-2.2
[2023-09-05 11:58:14,721 INFO][24510][submissive-well-off-rotanev][MainThread] - core.schains.monitor.main:189 - Adding config task to the pool
[2023-09-05 11:58:14,721 INFO][24510][submissive-well-off-rotanev][MainThread] - core.schains.monitor.main:157 - Monitor iteration completed, sleeping for 28
[2023-09-05 11:58:43,696 INFO][24510][submissive-well-off-rotanev][MainThread] - skale.contracts.manager.node_rotation:115 - current_ts: 1693915116, schain_finish_ts: 1693742124
[2023-09-05 11:58:44,858 INFO][24510][submissive-well-off-rotanev][MainThread] - core.schains.monitor.main:183 - sync_config_run False, config_version 0.0.0, stream_version regression-2.2
[2023-09-05 11:58:44,858 INFO][24510][submissive-well-off-rotanev][MainThread] - core.schains.monitor.main:189 - Adding config task to the pool
[2023-09-05 11:58:44,858 INFO][24510][submissive-well-off-rotanev][MainThread] - core.schains.monitor.main:157 - Monitor iteration completed, sleeping for 32
[2023-09-05 11:59:17,746 INFO][24510][submissive-well-off-rotanev][T_0] - skale.schain_config.rotation_history:40 - Collecting rotation history for submissive-well-off-rotanev...
[2023-09-05 11:59:17,851 INFO][24510][submissive-well-off-rotanev][MainThread] - skale.contracts.manager.node_rotation:115 - current_ts: 1693915152, schain_finish_ts: 1693742124
[2023-09-05 11:59:18,920 INFO][24510][submissive-well-off-rotanev][T_0] - skale.schain_config.rotation_history:50 - Rotation data for submissive-well-off-rotanev: Rotation(leaving_node_id=14, new_node_id=65, freeze_until=1693741836, rotation_counter=8)
[2023-09-05 11:59:18,996 INFO][24510][submissive-well-off-rotanev][MainThread] - core.schains.monitor.main:183 - sync_config_run False, config_version 0.0.0, stream_version regression-2.2
[2023-09-05 11:59:18,997 INFO][24510][submissive-well-off-rotanev][MainThread] - core.schains.monitor.main:189 - Adding config task to the pool
[2023-09-05 11:59:18,997 INFO][24510][submissive-well-off-rotanev][MainThread] - core.schains.monitor.main:157 - Monitor iteration completed, sleeping for 28
[2023-09-05 11:59:35,885 INFO][24510][submissive-well-off-rotanev][T_0] - skale.schain_config.rotation_history:142 - Adding rotation: 63 -> 65
[2023-09-05 11:59:41,591 INFO][24510][submissive-well-off-rotanev][T_0] - skale.schain_config.rotation_history:142 - Adding rotation: 23 -> 24
[2023-09-05 11:59:46,709 INFO][24510][submissive-well-off-rotanev][T_0] - skale.schain_config.rotation_history:142 - Adding rotation: 9 -> 64
[2023-09-05 11:59:47,979 INFO][24510][submissive-well-off-rotanev][MainThread] - skale.contracts.manager.node_rotation:115 - current_ts: 1693915176, schain_finish_ts: 1693742124
[2023-09-05 11:59:49,132 INFO][24510][submissive-well-off-rotanev][MainThread] - core.schains.monitor.main:183 - sync_config_run False, config_version 0.0.0, stream_version regression-2.2
[2023-09-05 11:59:49,133 INFO][24510][submissive-well-off-rotanev][MainThread] - core.schains.monitor.main:189 - Adding config task to the pool
[2023-09-05 11:59:49,133 INFO][24510][submissive-well-off-rotanev][MainThread] - core.schains.monitor.main:157 - Monitor iteration completed, sleeping for 33
[2023-09-05 11:59:52,244 INFO][24510][submissive-well-off-rotanev][T_0] - skale.schain_config.rotation_history:142 - Adding rotation: 35 -> 16
[2023-09-05 11:59:58,314 INFO][24510][submissive-well-off-rotanev][T_0] - skale.schain_config.rotation_history:142 - Adding rotation: 55 -> 63
[2023-09-05 12:00:04,393 INFO][24510][submissive-well-off-rotanev][T_0] - skale.schain_config.rotation_history:142 - Adding rotation: 14 -> 11
[2023-09-05 12:00:11,478 INFO][24510][submissive-well-off-rotanev][T_0] - skale.schain_config.rotation_history:142 - Adding rotation: 56 -> 60
```

```
[2023-09-05 12:00:18,301 ERROR][24510][submissive-well-off-rotanev][T_0] - core.schains.task:26 - Task submissive-well-off-rotanev-config failed with max() arg is an empty sequence
Traceback (most recent call last):
  File "/usr/src/admin/core/schains/task.py", line 24, in run
    self.action()
  File "/usr/src/admin/core/schains/monitor/main.py", line 105, in run_config_pipeline
    mon.run()
  File "/usr/src/admin/core/schains/monitor/config_monitor.py", line 48, in run
    self.execute()
  File "/usr/src/admin/core/schains/monitor/config_monitor.py", line 63, in execute
    self.am.upstream_config()
  File "/usr/src/admin/core/schains/monitor/action.py", line 96, in _monitor_block
    initial_status = f(self, *args, **kwargs)
  File "/usr/src/admin/core/schains/monitor/action.py", line 194, in upstream_config
    new_config = create_new_upstream_config(
  File "/usr/src/admin/core/schains/config/main.py", line 51, in create_new_upstream_config
    schain_config = generate_schain_config_with_skale(
  File "/usr/src/admin/core/schains/config/generator.py", line 213, in generate_schain_config_with_skale
    node_groups = get_previous_schain_groups(skale, schain_name)
  File "/usr/local/lib/python3.9/site-packages/skale/schain_config/rotation_history.py", line 56, in get_previous_schain_groups
    _add_previous_schain_rotations_state(
  File "/usr/local/lib/python3.9/site-packages/skale/schain_config/rotation_history.py", line 119, in _add_previous_schain_rotations_state
    new_node_id = max(previous_nodes.items(), key=lambda x: x[1]['finish_ts'])[0]
ValueError: max() arg is an empty sequence
```
[config_generation.txt](https://github.com/skalenetwork/skale-admin/files/12524542/config_generation.txt)

**Comment by badrogger** [comment]
restartSchainCreation transaction do not update _rotations mapping from skale-manager NodeRotation.sol.
That means that for the newly selected node there is no previous one. Therefore when config is generated it will be unable to compose the group

**Comment by PolinaKiporenko** [comment]
after discussion decided to unblock QA testing with recreation test chains and investigate and fix issue in 2.3 release https://github.com/skalenetwork/internal-support/issues/545
